Automattic/jetpack
# [RNMobile] Fetch VideoPress metadata for native block

Part of a series of PRs to address https://github.com/wordpress-mobile/gutenberg-mobile/issues/5611. 

## Related PRs

The changes in this PR relies on the following:

* `gutenberg-mobile`: https://github.com/wordpress-mobile/gutenberg-mobile/pull/5652

## Proposed changes:

* https://github.com/Automattic/jetpack/pull/29997/commits/06b3340d5223c4a23abb8a3a936846445178ffbe: Skip an unnecessary `isUserConnected` check. As only simple WordPress.com sites are supported on native for the time being, we can safely assume that users will be connected to Jetpack. We can revisit this in the future when expanding support. 
* https://github.com/Automattic/jetpack/pull/29997/commits/fe1ba4d1dbb4844d0ba6f90f97cb5c10ba9db198: To avoid creating native-specific files with very few changes, the `debug` functionality (which isn't compatible with native by default) is conditionally changed to `console.log`. 
* https://github.com/Automattic/jetpack/pull/29997/commits/aa4bf9b3244f3a7af9f72fc641be29732d008347: We pass the endpoint as a `path` on native, to ensure the URL is correctly built in the native apps and to allows us more control over supported endpoints. Support for this endpoint is added in https://github.com/wordpress-mobile/gutenberg-mobile/pull/5652.

### Other information:


## Jetpack product discussion

p9ugOq-3mV-p2 

## Does this pull request change what data or activity we track or use?

No, it does not.

## Testing instructions:

> **Note**
> The changes in this PR make sure the VideoPress metadata is synced from the web → app only. The necessary changes to ensure the metadata is also synced from the app → web will come in a future PR.

* As the `useSyncMedia` hook has not yet been implemented on native, it's necessary to first apply the following patch before testing these changes.

<details>
 <summary>Patch here ⤵️</summary>

These changes will implement the `useSyncMedia` hook and log the results of `useVideoData`: 

```patch
diff --git a/projects/packages/videopress/src/client/block-editor/blocks/video/edit.native.tsx b/projects/packages/videopress/src/client/block-editor/blocks/video/edit.native.tsx
index ff9906b1bf..ec61816006 100644
--- a/projects/packages/videopress/src/client/block-editor/blocks/video/edit.native.tsx
+++ b/projects/packages/videopress/src/client/block-editor/blocks/video/edit.native.tsx
@@ -13,6 +13,7 @@ import { useDispatch, useSelect } from '@wordpress/data';
 import { useState, useCallback, useEffect } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
 import { store as noticesStore } from '@wordpress/notices';
+import { useSyncMedia } from '../../hooks/use-sync-media';
 /**
  * External dependencies
  */
@@ -133,6 +134,8 @@ export default function VideoPressEdit( {
 		[ setAttributes ]
 	);
 
+	useSyncMedia( attributes, setAttributes );
+
 	const handleDoneUpload = useCallback(
 		newVideoData => {
 			setIsUploadingFile( false );
diff --git a/projects/packages/videopress/src/client/block-editor/hooks/use-sync-media/index.ts b/projects/packages/videopress/src/client/block-editor/hooks/use-sync-media/index.ts
index 66ec465133..5c1a39dc9d 100644
--- a/projects/packages/videopress/src/client/block-editor/hooks/use-sync-media/index.ts
+++ b/projects/packages/videopress/src/client/block-editor/hooks/use-sync-media/index.ts
@@ -279,7 +279,7 @@ export function useSyncMedia(
 	 * Store and compare the block attributes
 	 * in order to detect changes on them.
 	 */
-	const { isGeneratingPoster } = useVideoPosterData( attributes );
+	const isGeneratingPoster = false;
 
 	/*
 	 * Block attributes => Media data (sync)
diff --git a/projects/packages/videopress/src/client/block-editor/hooks/use-video-data/index.ts b/projects/packages/videopress/src/client/block-editor/hooks/use-video-data/index.ts
index f5b3b0816c..22c29c21b7 100644
--- a/projects/packages/videopress/src/client/block-editor/hooks/use-video-data/index.ts
+++ b/projects/packages/videopress/src/client/block-editor/hooks/use-video-data/index.ts
@@ -63,6 +63,10 @@ export default function useVideoData( {
 						skipRatingControl,
 					} );
 
+					const responseString = JSON.stringify( response );
+
+					console.log( 'video data: ' + responseString );
+
 					if ( response.duration ) {
 						debug(
 							`video duration available: ${ response.duration }, retried ${ retries } times`,
```
</details>

* With the patch applied, first navigate to the post editor on the web.
* Either add a new VideoPress block or use an existing one, and make sure to tweak its settings.
* It's particularly important to tweak the settings in the **Details** and **Privacy and rating** panels,  as these are the settings that are not currently being synced.
* Save your change on the web and then navigate to the post in the native app.
* Verify that the block's settings have been correctly fetched from the web.
* Verify also that `video data: {video data}` has been correctly logged to the console.
* Go through a few tries of editing the block on the web then returning the app to ensure syncing is working from the web → app.
